pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,304 @@
+	usage += "  # Set a log verbosity level to multiple components:\n"
+	usage += "  {{ProgramName}} adm logVerbosity 3 --virt-api  --virt-handler"
+	usage += "  # Set a log verbosity level to all components:\n"
+	usage += "  {{ProgramName}} adm logVerbosity 3 --all"
+	usage += "  # Reset the log verbosity level to all components:\n"
+	usage += "  {{ProgramName}} adm logVerbosity --reset"
+
+	return usage
+}
+
+func (c *Command) RunE(args []string) error {
+	/*
+	 * flag strength: reset > all > component
+	 */
+
+	var logLevel uint
+
+	// parse argument
+	if !resetFlag {
+		if len(args) == 0 {
+			return fmt.Errorf("expecting log verbosity level")
+		}
+		level, err := strconv.ParseUint(args[0], 10, 0)
+		if err != nil {
+			return err
+		}
+		logLevel = uint(level)
+
+		/*
+		 * user can specify logLevel 0
+		 * if we do not allow user to specify logLevel 0, need error handling here.
+		 */
+
+	}
+
+	// adm command always sees the KubeVirt resource of namespace="kubevirt", name="kubevirt"
+	const (
+		NAMESPACE = "kubevirt"
+		NAME      = "kubevirt"
+	)
+
+	virtClient, err := kubecli.GetKubevirtClientFromClientConfig(c.clientConfig)
+	if err != nil {
+		return fmt.Errorf("cannot obtain KubeVirt client: %v", err)
+	}
+
+	// get KubeVirt object
+	kv, err := virtClient.KubeVirt(NAMESPACE).Get(NAME, &k8smetav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// create patch
+	var patch string
+
+	// get LogVerbosity strcut entries
+	lv := kv.Spec.Configuration.DeveloperConfiguration.LogVerbosity
+
+	if lv == nil {
+		// no LogVerbosity etnry yet
+
+		if resetFlag {
+
+			// do nothing, because there exist no LogVerbosity setting
+			return nil
+
+		} else if allFlag {
+
+			patch = fmt.Sprintf(`
+					[{"op": "add","path": "/spec/configuration/developerConfiguration/logVerbosity","value": {"virtHandler": %d, "virtAPI": %d, "virtController": %d, "virtLauncher": %d, "virtOperator": %d}}]
+				`, logLevel, logLevel, logLevel, logLevel, logLevel)
+
+		} else if virtAPIFlag || virtControllerFlag || virtHanderFlag || virtLauncherFlag || virtOperatorFlag {
+
+			var components []string
+
+			if virtAPIFlag {
+				components = append(components, "virtAPI")
+			}
+			if virtControllerFlag {
+				components = append(components, "virtController")
+			}
+			if virtHanderFlag {
+				components = append(components, "virtHandler")
+			}
+			if virtLauncherFlag {
+				components = append(components, "virtLauncher")
+			}
+			if virtOperatorFlag {
+				components = append(components, "virtOperator")
+			}
+
+			// components never nil
+
+			patch = fmt.Sprintf(`[{"op": "add","path": "/spec/configuration/developerConfiguration/logVerbosity","value": {`)
+			for i, component := range components {
+				if i == (len(components) - 1) {
+					patch += fmt.Sprintf(`"%s": %d`, component, logLevel)
+				} else {
+					patch += fmt.Sprintf(`"%s": %d,`, component, logLevel)
+				}
+			}
+			patch += fmt.Sprintf(`}}]`)
```

Comment:
It should be possible to use the LogVerbosity struct and json.Encode() it, see a sample in the [go-playgound](https://play.golang.com/p/Dth1Vz-QEap)

Reply:
Note that thanks to `omitempty` you can probably relax some of the logic here.

Reply:
I understand how to handle nil value. Thank you for providing me with a guidance and sample code. This helps a lot.